driver.go

```diff
@@ -357,89 +353,118 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
+		cpuShares := uint64(cfg.Resources.LinuxResources.CPUShares)
+		createOpts.ContainerResourceConfig.ResourceLimits.CPU.Shares = &cpuShares
+	}
+	// -------------------------------------------------------------------------------------------
+	// SECURITY
+	// -------------------------------------------------------------------------------------------
+	createOpts.ContainerSecurityConfig.CapAdd = driverConfig.CapAdd
+	createOpts.ContainerSecurityConfig.CapDrop = driverConfig.CapDrop
+	createOpts.ContainerSecurityConfig.User = cfg.User
+
+	// -------------------------------------------------------------------------------------------
+	// NETWORK
+	// -------------------------------------------------------------------------------------------
+	for _, strdns := range driverConfig.Dns {
+		ipdns := net.ParseIP(strdns)
+		if ipdns == nil {
+			return nil, nil, fmt.Errorf("Invald dns server address")
 		}
+		createOpts.ContainerNetworkConfig.DNSServers = append(createOpts.ContainerNetworkConfig.DNSServers, ipdns)
 	}
-
-	// Generate network string
-	var network string
-	if cfg.NetworkIsolation != nil &&
-		cfg.NetworkIsolation.Path != "" {
-		network = fmt.Sprintf("ns:%s", cfg.NetworkIsolation.Path)
+	// Configure network
+	if cfg.NetworkIsolation != nil && cfg.NetworkIsolation.Path != "" {
+		createOpts.ContainerNetworkConfig.NetNS.NSMode = apiclient.Path
+		createOpts.ContainerNetworkConfig.NetNS.Value = cfg.NetworkIsolation.Path
 	} else {
-		network = driverConfig.NetworkMode
-	}
-
-	createOpts := iopodman.Create{
-		Args:              allArgs,
-		Entrypoint:        entryPoint,
-		WorkDir:           workingDir,
-		Env:               &allEnv,
-		Name:              &containerName,
-		Volume:            &allVolumes,
-		Memory:            &memoryLimit,
-		CpuShares:         &cpuShares,
-		CapAdd:            &driverConfig.CapAdd,
-		CapDrop:           &driverConfig.CapDrop,
-		Dns:               &driverConfig.Dns,
-		LogOpt:            &logOpts,
-		Hostname:          &driverConfig.Hostname,
-		Init:              &driverConfig.Init,
-		InitPath:          &driverConfig.InitPath,
-		User:              &cfg.User,
-		MemoryReservation: &driverConfig.MemoryReservation,
-		MemorySwap:        &swap,
-		MemorySwappiness:  swappiness,
-		Network:           &network,
-		Tmpfs:             &driverConfig.Tmpfs,
+		if driverConfig.NetworkMode == "" {
+			createOpts.ContainerNetworkConfig.NetNS.NSMode = apiclient.Bridge
+		} else if driverConfig.NetworkMode == "bridge" {
+			createOpts.ContainerNetworkConfig.NetNS.NSMode = apiclient.Bridge
+		} else if driverConfig.NetworkMode == "host" {
+			createOpts.ContainerNetworkConfig.NetNS.NSMode = apiclient.Host
+		} else if driverConfig.NetworkMode == "none" {
+			createOpts.ContainerNetworkConfig.NetNS.NSMode = apiclient.NoNetwork
+		} else if driverConfig.NetworkMode == "slirp4netns" {
+			createOpts.ContainerNetworkConfig.NetNS.NSMode = apiclient.Slirp
+		} else {
+			// FIXME: needs more work, parsing etc.
```

Comment:
can we track this with a github issue

Reply:
This "else" part is a typical place to handle not-yet-known requirements. I would not know what to write into the issue except "implement all possible podman network modes and switches".  We can remove the FIXME because it's not a bug per se.